### PR TITLE
Pyramid split fix

### PR DESCRIPTION
**Pull request by janos**
This PR: 

- addresses issue described here https://github.com/ethersphere/go-ethereum/issues/639
- changes pyramid splitting to handle edge cases where there is only one data chunk on the last branch, while preserving append and compress functionalities
- adds more file sizes in TestRandomData storage package test
- adds TestLocalStoreAndRetrieve test in swarm package to check more edge cases and chunk integration with Swarm

### CODE REVIEW

**Review by nonsense — APPROVED** [review]

**Review by gbalint — APPROVED** [review]
Thanks for comments!

File: swarm/swarm_test.go [hunk 1]
```diff
@@ -274,3 +277,82 @@ func TestParseEnsAPIAddress(t *testing.T) {
 		})
 	}
 }
+
+// TestLocalStoreAndRetrieve runs multiple tests where different size files are uploaded
+// to a single Swarm instance using API Store and checked against the dontent returned
```
Comment (zelig) [inline swarm/swarm_test.go]: dontent -> content

**Review by zelig — APPROVED** [review]

**Comment by janos** [comment]
Thank you for reviews. :) I have updated the test and fixed comments, travis is green, merging.
